routes/index.js

```diff
@@ -7,59 +7,74 @@ var Web3 = require('web3');
+      const name = parts[0] + '.eth';
       const address = await web3.eth.ens.getAddress(name);
-      const abi = JSON.parse((await axios.get('https://api.etherscan.io/api?module=contract&action=getabi&address='+address)).data.result);
+      const abi = JSON.parse(
+        (
+          await axios.get(
+            'https://api.etherscan.io/api?module=contract&action=getabi&address=' + address,
+          )
+        ).data.result,
+      );
       const hash = await storeAbi(abi);
 
-      return res.redirect('/'+address+"?abi="+hash);	    
+      return res.redirect('/' + address + '?abi=' + hash);
     }
-  }
-  catch(e){
-	  console.log(e);
-	  return res.render('error');
+  } catch (e) {
+    console.log(e);
+    return res.render('error');
   }
   res.render('new', { title: 'Express' });
 });
 
-router.get('/new', async function(req, res){
+router.get('/new', async function (req, res) {
   res.render('new');
-})
+});
 
-router.post('/new', async (req, res)=> {
+router.post('/new', async (req, res) => {
   const hash = await storeAbi(JSON.parse(req.body.abi));
-  res.redirect('/'+req.body.address+"/?abi="+hash);
-})
+  res.redirect('/' + req.body.address + '/?abi=' + hash + '&network=' + req.body.network);
+});
 
-router.get('/:address', async function(req, res) {
+router.get('/:address', async function (req, res) {
   const abiJson = await retrieveAbi(req.query.abi);
+  const network = req.query.network;
   let title = req.params.address;
-  const parts = req.hostname.split(".");
-  if(parts.length === 3){
-    title += " ("+parts[0]+".eth)";
+  const parts = req.hostname.split('.');
+  if (parts.length === 3) {
+    title += ' (' + parts[0] + '.eth)';
   }
 
-  const functions = abiJson.filter(a => a.type === "function");
-  const variables = abiJson.filter(a => a.type !== "function" && a.type !== "constructor");
-  res.render('index',  { functions, variables, address: req.params.address, abiEncoded: req.query.abi, title });
+  const functions = abiJson.filter((a) => a.type === 'function');
+  const variables = abiJson.filter((a) => a.type !== 'function' && a.type !== 'constructor');
+  res.render('index', {
+    functions,
+    variables,
+    address: req.params.address,
+    abiEncoded: req.query.abi,
+    title,
+    network,
+  });
 });
 
-router.get('/:address/:function', async function(req, res) {
+router.get('/:address/:function', async function (req, res) {
   try {
-  const abiJson = await retrieveAbi(req.query.abi);
-  const fun = abiJson.filter(a => a.type === "function" && a.name === req.params.function)[0];
-  res.render('function',  { fun, address: req.params.address, abiJson, abiEncoded: req.query.abi });
-  }
-  catch(e){
+    const abiJson = await retrieveAbi(req.query.abi);
+    const fun = abiJson.filter((a) => a.type === 'function' && a.name === req.params.function)[0];
+    res.render('function', {
+      fun,
+      address: req.params.address,
+      abiJson,
+      abiEncoded: req.query.abi,
```

Comment:
We will need network here too.

Reply:
Changes
- Added a common template module for wallet setup and checking and switching to correct network
- refactored the code to use ethers.js library. 
- added some more settings for default network of metamask as well so that even though the network is added by default the user still gets the option to switch to correct network via button.